### PR TITLE
TM-785 - allow run unit tests without argv

### PR DESCRIPTION
**Pull request by marceloandrader**
and allow set a profile on authenticator, this changes are needed in order the branch https://bitbucket.org/s_c/tm2-api/pull-requests/171/tm-785/diff works correctly.

### CODE REVIEW

File: src/Authentication/Authenticator.php [hunk 1]
```diff
@@ -134,6 +134,17 @@ function getProfile()
...
     /**
+     * will return a valid userProfile object for a pre-loaded profile
+     * (non-PHPdoc)
+     *
+     * @see \PhalconRest\Authentication\AuthenticatorInterface::getProfile()
+     */
+    function setProfile(\PhalconRest\Authentication\UserProfile $profile)
+    {
+        return $this->profile = $profile;
+    }
```
Comment (igorsantos07) [inline src/Authentication/Authenticator.php:145]: :thinking: does it make sense to add this directly?
The design of this class seems to mean the profile should not be touched directly, but instead changed from the inside through the `authenticate()` method, making sure it always have valid data and thus, making this class a trusty one.

The usage of this on the Bitbucket PR also seems a bit weird...

Comment (igorsantos07) [reply src/Authentication/Authenticator.php:145]: well, turns out this is not needed and the "hole" I envisioned is already present on the API. You can use loadProfile() from the Authenticator component to achieve the same result without adding a new method that overrides the already created object :)

Remove this method and I'll merge the PR.

Will explain @ Bitbucket.

**Review by igorsantos07 — CHANGES_REQUESTED** [review]
This commit seems to have caused a conflict. Try to rebase and fix it.

File: src/Authentication/Authenticator.php (hunk 1 above)
Comment (igorsantos07) [reply src/Authentication/Authenticator.php:145]: BTW, would you also change this bit of documentation at `libraries/Authentication/UserProfile::loadProfile`? The method signature changed from the original method and the IDE is a bit dizzy on that.

I can't add it directly to this PR as I would need to fork your fork to send you a PR to be included in this PR.... lol

```php
    /**
     * load a full user profile based on three different filters
     *
     * ie.
     * "token = 'HACKYHACKERSON'"
     * "user_name = 'jjenkins'"
     * "token = 'A REALLY LONG STRING"
     *
     *
     * (non-PHPdoc)
     *
     * @see \PhalconRest\Authentication\UserProfile::loadProfileByToken()
     * @param string $mainFilter
     * @return bool
     * @throws HTTPException
     */
```

**Comment by marceloandrader** [comment]
I'll create a new one.

**Pull request closed without merging**